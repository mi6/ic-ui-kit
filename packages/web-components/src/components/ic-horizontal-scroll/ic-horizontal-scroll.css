@import "../../global/normalize.css";

:host {
  --border-bottom-color: var(--ic-architectural-300);
  --splitter-color: var(--ic-architectural-100);

  display: flex;
  max-width: 100%;
}

:host(.light) {
  --splitter-color: var(--ic-keyline-lighten-rgb);
}

:host(.dark) {
  --splitter-color: var(--ic-keyline-darken-rgb);
}

/* Slotted items container element */
::slotted(*) {
  overflow-x: scroll;
  overflow-y: hidden;
  scroll-behavior: smooth;
  scrollbar-width: none;
  -ms-overflow-style: none;
  margin: calc(var(--ic-space-xs) * -1) !important;
  padding: var(--ic-space-xs) !important;
}

:host(.visible) ::slotted(*) {
  margin: calc(var(--ic-space-xs) * -1) 0 !important;
  padding: var(--ic-space-xs) !important;
}

.tabs-container::-webkit-scrollbar {
  display: none;
}

.scroll-container-left,
.scroll-container-right {
  display: flex;
}

.scroll-container-left {
  margin-left: var(--ic-space-xxs);
}

.scroll-container-right {
  margin-right: var(--ic-space-xxs);
}

.hidden {
  display: none;
}

.scroll-arrow {
  margin: var(--ic-space-xxs) 0 var(--ic-space-xs) 0;
}

.disabled .ic-tooltip-container {
  display: none !important;
}

.scroll-splitter-left,
.scroll-splitter-right {
  display: flex;
  height: var(--ic-space-lg);
  width: var(--ic-space-1px);
  margin-top: var(--ic-space-xs);
  border-radius: var(--ic-space-1px);
  background-color: var(--splitter-color);
}

.scroll-splitter-left {
  margin-left: calc(var(--ic-space-xxs) - var(--ic-space-1px));
}

.scroll-splitter-right {
  margin-right: calc(var(--ic-space-xxs) - var(--ic-space-1px));
}

.disabled .scroll-splitter-left,
.disabled .scroll-splitter-right {
  background-color: transparent;
}

.disabled ic-button > svg {
  color: var(--ic-keyline-darken-rgb);
}

:host(.dark) .disabled ic-button > svg,
:host(.light) .disabled ic-button > svg {
  color: var(--splitter-color) !important;
}
