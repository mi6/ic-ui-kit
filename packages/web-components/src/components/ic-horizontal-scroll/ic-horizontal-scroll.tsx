import {
  Component,
  Element,
  h,
  Host,
  State,
  Method,
  Prop,
} from "@stencil/core";

import LeftArrow from "./assets/left-arrow.svg";
import RightArrow from "./assets/right-arrow.svg";
import { getSlotElements } from "../../utils/helpers";

import { checkResizeObserver, elementOverflowsX } from "../../utils/helpers";
import { IcThemeForeground, IcThemeForegroundEnum } from "../../utils/types";

const SCROLL_DELAY_MS = 200;

@Component({
  tag: "ic-horizontal-scroll",
  styleUrl: "./ic-horizontal-scroll.css",
  shadow: true,
})
export class HorizontalScroll {
  private buttonStateSet: boolean = false;
  private isScrolling: number;
  private itemOffsets: number[];
  private items: HTMLElement[];
  private itemsContainerEl: HTMLElement;
  private resizeObserver: ResizeObserver;
  private scrollDelay: number;

  @Element() el: HTMLIcHorizontalScrollElement;

  @State() firstItemVisible: boolean = true;
  @State() itemOverflow: boolean = false;
  @State() lastItemVisible: boolean = false;

  /**
   * The appearance of the horizontal scroll, e.g. dark, light or the default.
   */
  @Prop() appearance?: IcThemeForeground = "default";

  componentWillLoad(): void {
    this.itemsContainerEl = this.el.children[0] as HTMLElement;
    this.itemsContainerEl.addEventListener("scroll", this.scrollHandler);

    this.items = (getSlotElements(this.itemsContainerEl) ||
      Array.from(this.itemsContainerEl.children)) as HTMLElement[];
    this.items.forEach((item) => {
      if (item.addEventListener) {
        item.addEventListener("focus", this.focusHandler);
      }
    });
  }

  componentDidLoad(): void {
    let runningTotal = 0;
    this.itemOffsets = this.items.map((item) => {
      runningTotal += item.offsetWidth;
      return runningTotal;
    });

    checkResizeObserver(this.runResizeObserver);

    // Add event listener to scroll containers as mouse events are not fired on disabled elements (ic-button's <button>)
    // 'mouseleave' needed in case the user moves their mouse while holding the arrow buttons
    // - 'mouseup' otherwise not detected and scrolling not stopped
    const scrollArrows = Array.from(
      this.el.shadowRoot.querySelectorAll("div")
    ) as HTMLElement[];
    ["mouseup", "mouseleave"].forEach((event) => {
      scrollArrows.forEach((arrow) =>
        arrow.addEventListener(event, this.arrowMouseUpHandler)
      );
    });
  }

  disconnectedCallback(): void {
    if (this.resizeObserver !== undefined) {
      this.resizeObserver.disconnect();
    }

    const scrollArrows = Array.from(
      this.el.shadowRoot.querySelectorAll("div")
    ) as HTMLElement[];
    ["mouseup", "mouseleave"].forEach((event) => {
      scrollArrows.forEach((arrow) =>
        arrow.removeEventListener(event, this.arrowMouseUpHandler)
      );
    });

    this.items.forEach((item) => {
      if (item.removeEventListener) {
        item.removeEventListener("focus", this.focusHandler);
      }
    });

    this.itemsContainerEl.removeEventListener("scroll", this.scrollHandler);
  }

  /**
   * @internal if side scrolling enabled, scrolls the specified item into view.
   */
  @Method()
  async scrollItemIntoView(itemPosition: number): Promise<void> {
    this.firstItemVisible = itemPosition <= 0;
    const newScrollPos =
      itemPosition <= 0 ? 0 : this.itemOffsets[itemPosition - 1];
    this.lastItemVisible =
      this.itemsContainerEl.offsetWidth + newScrollPos >=
      this.itemsContainerEl.scrollWidth;
    this.buttonStateSet = true;
    this.itemsContainerEl.scrollLeft = newScrollPos;
  }

  private focusHandler = (event: FocusEvent) => {
    this.itemFocusHandler(
      Array.from(this.items).indexOf(event.target as HTMLElement)
    );
  };

  private itemFocusHandler(itemPosition: number): void {
    if (this.itemOverflow) {
      this.scrollItemIntoView(itemPosition);
    }
  }

  private resizeObserverCallback = () => {
    if (this.el.clientWidth >= this.itemsContainerEl.scrollWidth) {
      this.itemOverflow = false;
    } else {
      this.itemOverflow = elementOverflowsX(this.itemsContainerEl);
    }

    if (this.itemOverflow) {
      this.lastItemVisible =
        this.itemsContainerEl.offsetWidth + this.itemsContainerEl.scrollLeft >=
        this.itemsContainerEl.scrollWidth;
    }
  };

  private runResizeObserver = () => {
    this.resizeObserver = new ResizeObserver(() => {
      this.resizeObserverCallback();
    });
    this.resizeObserver.observe(this.itemsContainerEl);
  };

  private scrollLeft = () => {
    this.scrollItemIntoView(this.getCurrentLeftItem() - 1);
  };

  private scrollRight = () => {
    this.scrollItemIntoView(this.getCurrentLeftItem() + 1);
  };

  private longScrollRight = () => {
    this.scrollRight();
    this.scrollDelay = window.setTimeout(this.longScrollRight, SCROLL_DELAY_MS);
  };

  private longScrollLeft = () => {
    this.scrollLeft();
    this.scrollDelay = window.setTimeout(this.longScrollLeft, SCROLL_DELAY_MS);
  };

  private leftArrowMouseDownHandler = (e: MouseEvent) => {
    e.preventDefault();
    this.scrollDelay = window.setTimeout(this.longScrollLeft, SCROLL_DELAY_MS);
  };

  private rightArrowMouseDownHandler = (e: MouseEvent) => {
    e.preventDefault();
    this.scrollDelay = window.setTimeout(this.longScrollRight, SCROLL_DELAY_MS);
  };

  private arrowMouseUpHandler = () => {
    window.clearTimeout(this.scrollDelay);
  };

  private getCurrentLeftItem = (): number => {
    const index = this.itemOffsets.findIndex(
      (el) => el > Math.round(this.itemsContainerEl.scrollLeft)
    );
    return index < 0 ? 0 : index;
  };

  private scrollHandler = () => {
    window.clearTimeout(this.isScrolling);
    // Set a timeout to run after scrolling ends
    this.isScrolling = window.setTimeout(this.scrollStopped, 50);
  };

  private scrollStopped = () => {
    // If scrollItemIntoView has been called, the button states will already be set
    // Can't just handle it here as it causes strange jumping behaviour in positioning
    const scrollLeft = Math.round(this.itemsContainerEl.scrollLeft);
    if (this.buttonStateSet === false) {
      this.firstItemVisible = scrollLeft === 0;
      this.lastItemVisible =
        this.itemsContainerEl.offsetWidth + scrollLeft >=
        this.itemsContainerEl.scrollWidth;
    }
    this.buttonStateSet = false;
  };

  render() {
    const { appearance, firstItemVisible, lastItemVisible, itemOverflow } =
      this;

    return (
      <Host
        class={{
          ["visible"]: itemOverflow,
          ["dark"]: this.appearance === IcThemeForegroundEnum.Dark,
          ["light"]: this.appearance === IcThemeForegroundEnum.Light,
        }}
      >
        <div
          aria-hidden="true"
          class={{
            ["scroll-container-left"]: true,
            ["hidden"]: !itemOverflow,
            ["disabled"]: firstItemVisible,
          }}
          role="tab"
        >
          <ic-button
            class="scroll-arrow"
            variant="icon"
            aria-label="Scroll left"
            appearance={appearance}
            innerHTML={LeftArrow}
            disabled={firstItemVisible}
            tabindex="-1"
            onClick={this.scrollLeft}
            onMouseDown={this.leftArrowMouseDownHandler}
          ></ic-button>
          <span class="scroll-splitter-left"></span>
        </div>
        <slot></slot>
        <div
          aria-hidden="true"
          class={{
            ["scroll-container-right"]: true,
            ["hidden"]: !itemOverflow,
            ["disabled"]: lastItemVisible,
          }}
          role="tab"
        >
          <span class="scroll-splitter-right"></span>
          <ic-button
            class="scroll-arrow"
            variant="icon"
            aria-label="Scroll right"
            appearance={appearance}
            innerHTML={RightArrow}
            disabled={lastItemVisible}
            tabindex="-1"
            onClick={this.scrollRight}
            onMouseDown={this.rightArrowMouseDownHandler}
          ></ic-button>
        </div>
      </Host>
    );
  }
}
