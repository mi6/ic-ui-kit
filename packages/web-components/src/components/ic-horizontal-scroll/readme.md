# ic-horizontal-scroll



<!-- Auto Generated Below -->


## Properties

| Property     | Attribute    | Description                                                               | Type                             | Default     |
| ------------ | ------------ | ------------------------------------------------------------------------- | -------------------------------- | ----------- |
| `appearance` | `appearance` | The appearance of the horizontal scroll, e.g. dark, light or the default. | `"dark" \| "default" \| "light"` | `"default"` |


## Dependencies

### Used by

 - [ic-page-header](../ic-page-header)
 - [ic-tab-group](../ic-tab-group)
 - [ic-top-navigation](../ic-top-navigation)

### Depends on

- [ic-button](../ic-button)

### Graph
```mermaid
graph TD;
  ic-horizontal-scroll --> ic-button
  ic-button --> ic-loading-indicator
  ic-button --> ic-tooltip
  ic-loading-indicator --> ic-typography
  ic-tooltip --> ic-typography
  ic-page-header --> ic-horizontal-scroll
  ic-tab-group --> ic-horizontal-scroll
  ic-top-navigation --> ic-horizontal-scroll
  style ic-horizontal-scroll fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------


