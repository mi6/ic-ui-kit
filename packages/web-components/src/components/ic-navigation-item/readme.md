# ic-navigation-item



<!-- Auto Generated Below -->


## Properties

| Property         | Attribute        | Description                                                                                         | Type                                                                                                                                                                                     | Default     |
| ---------------- | ---------------- | --------------------------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------- |
| `download`       | `download`       | If `true`, the user can save the linked URL instead of navigating to it.                            | `boolean \| string`                                                                                                                                                                      | `false`     |
| `href`           | `href`           | The destination of the navigation item.                                                             | `string`                                                                                                                                                                                 | `""`        |
| `hreflang`       | `hreflang`       | The human language of the linked URL.                                                               | `string`                                                                                                                                                                                 | `undefined` |
| `label`          | `label`          | The label of the navigation item.                                                                   | `string`                                                                                                                                                                                 | `undefined` |
| `referrerpolicy` | `referrerpolicy` | How much of the referrer to send when following the link.                                           | `"" \| "no-referrer" \| "no-referrer-when-downgrade" \| "origin" \| "origin-when-cross-origin" \| "same-origin" \| "strict-origin" \| "strict-origin-when-cross-origin" \| "unsafe-url"` | `undefined` |
| `rel`            | `rel`            | The relationship of the linked URL as space-separated link types.                                   | `string`                                                                                                                                                                                 | `undefined` |
| `selected`       | `selected`       | If `true`, the navigation item will be set in a selected state.                                     | `boolean`                                                                                                                                                                                | `false`     |
| `target`         | `target`         | The place to display the linked URL, as the name for a browsing context (a tab, window, or iframe). | `string`                                                                                                                                                                                 | `undefined` |


## Methods

### `setFocus() => Promise<void>`

Sets focus on the nav item.

#### Returns

Type: `Promise<void>`




## Shadow Parts

| Part     | Description                         |
| -------- | ----------------------------------- |
| `"link"` | The `<a>` within ic-navigation-item |


## CSS Custom Properties

| Name                           | Description                |
| ------------------------------ | -------------------------- |
| `--ic-z-index-navigation-item` | z-index of navigation item |


## Dependencies

### Depends on

- [ic-typography](../ic-typography)
- [ic-tooltip](../ic-tooltip)

### Graph
```mermaid
graph TD;
  ic-navigation-item --> ic-typography
  ic-navigation-item --> ic-tooltip
  ic-tooltip --> ic-typography
  style ic-navigation-item fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------


