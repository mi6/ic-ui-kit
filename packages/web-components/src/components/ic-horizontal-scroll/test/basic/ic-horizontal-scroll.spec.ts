import { newSpecPage } from "@stencil/core/testing";
import { NavigationItem } from "../../../ic-navigation-item/ic-navigation-item";
import { HorizontalScroll } from "../../ic-horizontal-scroll";

const testHorizontalScroll = `<ic-horizontal-scroll>
  <div>
    <ic-navigation-item label="Test nav item 1"></ic-navigation-item>
    <ic-navigation-item label="Test nav item 2"></ic-navigation-item>
    <ic-navigation-item label="Test nav item 3"></ic-navigation-item>
    <ic-navigation-item label="Test nav item 4"></ic-navigation-item>
    <ic-navigation-item label="Test nav item 5"></ic-navigation-item>
    <ic-navigation-item label="Test nav item 6"></ic-navigation-item>
  </div>
  </ic-horizontal-scroll>`;

describe("ic-horizontal-scroll", () => {
  it("should render", async () => {
    const page = await newSpecPage({
      components: [HorizontalScroll, NavigationItem],
      html: testHorizontalScroll,
    });

    expect(page.root).toMatchSnapshot("renders");
  });

  it("should render the correct HTML for the chosen appearance", async () => {
    const page = await newSpecPage({
      components: [HorizontalScroll, NavigationItem],
      html: testHorizontalScroll,
    });

    expect(page.root).toMatchSnapshot("renders-appearance");
  });

  it("should scroll item into view", async () => {
    const page = await newSpecPage({
      components: [HorizontalScroll, NavigationItem],
      html: testHorizontalScroll,
    });

    page.rootInstance.scrollItemIntoView(6);
    expect(page.rootInstance.buttonStateSet).toBe(true);
  });

  it("should long scroll left", async () => {
    const page = await newSpecPage({
      components: [HorizontalScroll, NavigationItem],
      html: testHorizontalScroll,
    });

    page.rootInstance.longScrollLeft();
    page.waitForChanges();

    expect(page.rootInstance.scrollDelay._idleNext.msecs).toBe(200);
  });

  it("should long scroll right", async () => {
    const page = await newSpecPage({
      components: [HorizontalScroll, NavigationItem],
      html: testHorizontalScroll,
    });

    page.rootInstance.longScrollRight();
    page.waitForChanges();

    expect(page.rootInstance.scrollDelay._idleNext.msecs).toBe(200);
  });

  it("should delay on left arrow mouse down and set a timeout after the scrolling ends", async () => {
    const page = await newSpecPage({
      components: [HorizontalScroll, NavigationItem],
      html: testHorizontalScroll,
    });

    const event = new Event("mousedown", {
      bubbles: true,
      cancelable: true,
    });

    page.rootInstance.leftArrowMouseDownHandler(event);
    page.waitForChanges();

    expect(page.rootInstance.scrollDelay._idleNext.msecs).toBe(200);

    page.rootInstance.isScrolling = 200;
    page.waitForChanges();

    page.rootInstance.scrollHandler();

    expect(page.rootInstance.isScrolling._idleNext.msecs).toBe(50);
  });

  it("should delay on right arrow mouse down and revoke delay on mouse up", async () => {
    const page = await newSpecPage({
      components: [HorizontalScroll, NavigationItem],
      html: testHorizontalScroll,
    });

    const event = new Event("mousedown", {
      bubbles: true,
      cancelable: true,
    });

    page.rootInstance.rightArrowMouseDownHandler(event);
    page.waitForChanges();

    expect(page.rootInstance.scrollDelay._idleNext.msecs).toBe(200);

    page.rootInstance.arrowMouseUpHandler();
    expect(page.rootInstance.scrollDelay._idleNext).toBeNull();
  });

  it("should stop scrolling if mouse leaves the scroll button", async () => {
    const page = await newSpecPage({
      components: [HorizontalScroll, NavigationItem],
      html: testHorizontalScroll,
    });

    const scrollContainer = page.root.shadowRoot.querySelector(
      ".scroll-container-right"
    );
    const mouseDownEvent = new Event("mousedown", {
      bubbles: true,
      cancelable: true,
    });
    const mouseLeaveEvent = new Event("mouseleave", {
      bubbles: true,
      cancelable: true,
    });

    page.rootInstance.rightArrowMouseDownHandler(mouseDownEvent);
    page.waitForChanges();

    scrollContainer.dispatchEvent(mouseLeaveEvent);
    page.waitForChanges();

    expect(page.rootInstance.arrowMouseUpHandler()).toHaveBeenCalled;
  });

  it("should stop scrolling", async () => {
    const page = await newSpecPage({
      components: [HorizontalScroll, NavigationItem],
      html: testHorizontalScroll,
    });

    //when buttonStateSet is true
    page.rootInstance.buttonStateSet = true;
    page.rootInstance.scrollStopped();
    page.waitForChanges();

    expect(page.rootInstance.buttonStateSet).toBe(false);

    //when buttonStateSet is false
    page.rootInstance.itemsContainerEl.scrollLeft = 0;
    page.rootInstance.scrollStopped();
    page.waitForChanges();

    expect(page.rootInstance.firstItemVisible).toBe(true);
  });

  it("should handle item focus", async () => {
    const page = await newSpecPage({
      components: [HorizontalScroll, NavigationItem],
      html: testHorizontalScroll,
    });

    page.rootInstance.itemOverflow = true;
    page.waitForChanges();

    const item: HTMLElement = page.root.querySelector(
      "[label='Test nav item 4']"
    );
    item.focus();
    page.waitForChanges();

    expect(page.rootInstance.itemFocusHandler()).toHaveBeenCalled;
    expect(page.rootInstance.firstItemVisible).toBe(false);
  });

  it("should render scroll arrows when items overflow", async () => {
    const page = await newSpecPage({
      components: [HorizontalScroll, NavigationItem],
      html: `<ic-horizontal-scroll>
        <div>
        <ic-navigation-item label="Test nav item 1"></ic-navigation-item>
        <ic-navigation-item label="Test nav item 2"></ic-navigation-item>
        <ic-navigation-item label="Test nav item 3"></ic-navigation-item>
        <ic-navigation-item label="Test nav item 4"></ic-navigation-item>
        <ic-navigation-item label="Test nav item 5"></ic-navigation-item>
        <ic-navigation-item label="Test nav item 6"></ic-navigation-item>
        </div>
      </ic-horizontal-scroll>`,
    });

    page.rootInstance.itemOverflow = true;
    page.waitForChanges();
    const scroll = page.root.shadowRoot.querySelector(".scroll-arrow");

    expect(scroll).not.toBeNull();

    //call runResizeObserver
    page.rootInstance.runResizeObserver();

    //test disconnected callback
    page.setContent("");
  });

  it("should set itemOverflow to true where scrollWidth > clientWidth on the scroll container element", async () => {
    const page = await newSpecPage({
      components: [HorizontalScroll, NavigationItem],
      html: `<ic-horizontal-scroll>
        <div>
        <ic-navigation-item label="Test nav item 1"></ic-navigation-item>
        <ic-navigation-item label="Test nav item 2"></ic-navigation-item>
        <ic-navigation-item label="Test nav item 3"></ic-navigation-item>
        <ic-navigation-item label="Test nav item 4"></ic-navigation-item>
        <ic-navigation-item label="Test nav item 5"></ic-navigation-item>
        <ic-navigation-item label="Test nav item 6"></ic-navigation-item>
        </div>
      </ic-horizontal-scroll>`,
    });

    page.rootInstance.itemsContainerEl.scrollWidth = 100;
    page.rootInstance.itemsContainerEl.clientWidth = 50;
    page.rootInstance.resizeObserverCallback();
    page.waitForChanges();

    expect(page.rootInstance.itemOverflow).toBe(true);
  });
});
