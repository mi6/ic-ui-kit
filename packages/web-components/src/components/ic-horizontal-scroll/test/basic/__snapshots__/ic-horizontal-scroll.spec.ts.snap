// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ic-horizontal-scroll should render the correct HTML for the chosen appearance: renders-appearance 1`] = `
<ic-horizontal-scroll>
  <mock:shadow-root>
    <div aria-hidden="true" class="disabled hidden scroll-container-left" role="tab">
      <ic-button appearance="default" aria-label="Scroll left" class="scroll-arrow" disabled="" tabindex="-1" variant="icon">
        svg
      </ic-button>
      <span class="scroll-splitter-left"></span>
    </div>
    <slot></slot>
    <div aria-hidden="true" class="hidden scroll-container-right" role="tab">
      <span class="scroll-splitter-right"></span>
      <ic-button appearance="default" aria-label="Scroll right" class="scroll-arrow" tabindex="-1" variant="icon">
        svg
      </ic-button>
    </div>
  </mock:shadow-root>
  <div>
    <ic-navigation-item class="light navigation-item" label="Test nav item 1" role="listitem">
      <mock:shadow-root>
        <ic-tooltip class="tooltip-navigation-item" label="Test nav item 1" placement="right" target="navigation-item">
          <div aria-labelledby="ic-tooltip-navigation-item" class="link" tabindex="0">
            <ic-typography variant="label">
              Test nav item 1
            </ic-typography>
          </div>
        </ic-tooltip>
      </mock:shadow-root>
    </ic-navigation-item>
    <ic-navigation-item class="light navigation-item" label="Test nav item 2" role="listitem">
      <mock:shadow-root>
        <ic-tooltip class="tooltip-navigation-item" label="Test nav item 2" placement="right" target="navigation-item">
          <div aria-labelledby="ic-tooltip-navigation-item" class="link" tabindex="0">
            <ic-typography variant="label">
              Test nav item 2
            </ic-typography>
          </div>
        </ic-tooltip>
      </mock:shadow-root>
    </ic-navigation-item>
    <ic-navigation-item class="light navigation-item" label="Test nav item 3" role="listitem">
      <mock:shadow-root>
        <ic-tooltip class="tooltip-navigation-item" label="Test nav item 3" placement="right" target="navigation-item">
          <div aria-labelledby="ic-tooltip-navigation-item" class="link" tabindex="0">
            <ic-typography variant="label">
              Test nav item 3
            </ic-typography>
          </div>
        </ic-tooltip>
      </mock:shadow-root>
    </ic-navigation-item>
    <ic-navigation-item class="light navigation-item" label="Test nav item 4" role="listitem">
      <mock:shadow-root>
        <ic-tooltip class="tooltip-navigation-item" label="Test nav item 4" placement="right" target="navigation-item">
          <div aria-labelledby="ic-tooltip-navigation-item" class="link" tabindex="0">
            <ic-typography variant="label">
              Test nav item 4
            </ic-typography>
          </div>
        </ic-tooltip>
      </mock:shadow-root>
    </ic-navigation-item>
    <ic-navigation-item class="light navigation-item" label="Test nav item 5" role="listitem">
      <mock:shadow-root>
        <ic-tooltip class="tooltip-navigation-item" label="Test nav item 5" placement="right" target="navigation-item">
          <div aria-labelledby="ic-tooltip-navigation-item" class="link" tabindex="0">
            <ic-typography variant="label">
              Test nav item 5
            </ic-typography>
          </div>
        </ic-tooltip>
      </mock:shadow-root>
    </ic-navigation-item>
    <ic-navigation-item class="light navigation-item" label="Test nav item 6" role="listitem">
      <mock:shadow-root>
        <ic-tooltip class="tooltip-navigation-item" label="Test nav item 6" placement="right" target="navigation-item">
          <div aria-labelledby="ic-tooltip-navigation-item" class="link" tabindex="0">
            <ic-typography variant="label">
              Test nav item 6
            </ic-typography>
          </div>
        </ic-tooltip>
      </mock:shadow-root>
    </ic-navigation-item>
  </div>
</ic-horizontal-scroll>
`;

exports[`ic-horizontal-scroll should render: renders 1`] = `
<ic-horizontal-scroll>
  <mock:shadow-root>
    <div aria-hidden="true" class="disabled hidden scroll-container-left" role="tab">
      <ic-button appearance="default" aria-label="Scroll left" class="scroll-arrow" disabled="" tabindex="-1" variant="icon">
        svg
      </ic-button>
      <span class="scroll-splitter-left"></span>
    </div>
    <slot></slot>
    <div aria-hidden="true" class="hidden scroll-container-right" role="tab">
      <span class="scroll-splitter-right"></span>
      <ic-button appearance="default" aria-label="Scroll right" class="scroll-arrow" tabindex="-1" variant="icon">
        svg
      </ic-button>
    </div>
  </mock:shadow-root>
  <div>
    <ic-navigation-item class="light navigation-item" label="Test nav item 1" role="listitem">
      <mock:shadow-root>
        <ic-tooltip class="tooltip-navigation-item" label="Test nav item 1" placement="right" target="navigation-item">
          <div aria-labelledby="ic-tooltip-navigation-item" class="link" tabindex="0">
            <ic-typography variant="label">
              Test nav item 1
            </ic-typography>
          </div>
        </ic-tooltip>
      </mock:shadow-root>
    </ic-navigation-item>
    <ic-navigation-item class="light navigation-item" label="Test nav item 2" role="listitem">
      <mock:shadow-root>
        <ic-tooltip class="tooltip-navigation-item" label="Test nav item 2" placement="right" target="navigation-item">
          <div aria-labelledby="ic-tooltip-navigation-item" class="link" tabindex="0">
            <ic-typography variant="label">
              Test nav item 2
            </ic-typography>
          </div>
        </ic-tooltip>
      </mock:shadow-root>
    </ic-navigation-item>
    <ic-navigation-item class="light navigation-item" label="Test nav item 3" role="listitem">
      <mock:shadow-root>
        <ic-tooltip class="tooltip-navigation-item" label="Test nav item 3" placement="right" target="navigation-item">
          <div aria-labelledby="ic-tooltip-navigation-item" class="link" tabindex="0">
            <ic-typography variant="label">
              Test nav item 3
            </ic-typography>
          </div>
        </ic-tooltip>
      </mock:shadow-root>
    </ic-navigation-item>
    <ic-navigation-item class="light navigation-item" label="Test nav item 4" role="listitem">
      <mock:shadow-root>
        <ic-tooltip class="tooltip-navigation-item" label="Test nav item 4" placement="right" target="navigation-item">
          <div aria-labelledby="ic-tooltip-navigation-item" class="link" tabindex="0">
            <ic-typography variant="label">
              Test nav item 4
            </ic-typography>
          </div>
        </ic-tooltip>
      </mock:shadow-root>
    </ic-navigation-item>
    <ic-navigation-item class="light navigation-item" label="Test nav item 5" role="listitem">
      <mock:shadow-root>
        <ic-tooltip class="tooltip-navigation-item" label="Test nav item 5" placement="right" target="navigation-item">
          <div aria-labelledby="ic-tooltip-navigation-item" class="link" tabindex="0">
            <ic-typography variant="label">
              Test nav item 5
            </ic-typography>
          </div>
        </ic-tooltip>
      </mock:shadow-root>
    </ic-navigation-item>
    <ic-navigation-item class="light navigation-item" label="Test nav item 6" role="listitem">
      <mock:shadow-root>
        <ic-tooltip class="tooltip-navigation-item" label="Test nav item 6" placement="right" target="navigation-item">
          <div aria-labelledby="ic-tooltip-navigation-item" class="link" tabindex="0">
            <ic-typography variant="label">
              Test nav item 6
            </ic-typography>
          </div>
        </ic-tooltip>
      </mock:shadow-root>
    </ic-navigation-item>
  </div>
</ic-horizontal-scroll>
`;
